pkg/api/controllers/catalog/catalog.go

```diff
@@ -21,21 +26,98 @@ type CacheCleaner struct {
 	catalogClient        v3.CatalogInterface
 	projectCatalogClient v3.ProjectCatalogInterface
 	clusterCatalogClient v3.ClusterCatalogInterface
+	manager              *manager.Manager
 	debounce             func(func())
 }
 
 func Register(ctx context.Context, context *config.ScaledContext) {
+	managementCtx, err := context.NewManagementContext()
+	if err != nil {
+		panic(err)
+	}
 	cleaner := &CacheCleaner{
 		catalogClient:        context.Management.Catalogs(""),
 		projectCatalogClient: context.Management.ProjectCatalogs(""),
 		clusterCatalogClient: context.Management.ClusterCatalogs(""),
+		manager:              manager.New(managementCtx),
```

Comment:
This seems like a lot of change just for updating error state of a catalog, is there an easier way?

Reply:
Catalog manager is currently registered in `managementContext`, so either we want to move the code to the scaledContext controller or register a new context in the catalog controller(current way).

Reply:
Why do we need the catalog manager just to update the error state of the catalog?

Reply:
not only just for updating the error state, but most catalog function we want to call is also registered in the `managementContext`, so trying to avoid write duplicate codes and that why we want to register a new manager context in here.

Reply:
The only code we are calling from managementContext is UpdateCatalogError tho, everything else is done helmlib. It seems odd creating a new management context in a scaled context to do this.

Reply:
updated.

Reply:
@guangbochen I don't think this PR is needed anymore with #20198

Reply:
do you have a build test for https://github.com/rancher/rancher/pull/20198, I only see git fetch enhancement.

Reply:
Are you asking for a validation test?